evdns.c

```diff
@@ -731,9 +731,32 @@ nameserver_failed(struct nameserver *const ns, const char *msg) {
 	ns->state = 0;
 	ns->failed_times = 1;
 
-	disconnect_and_free_connection(ns->connection);
-	ns->connection = NULL;
+	if (ns->connection) {
+		disconnect_and_free_connection(ns->connection);
+		ns->connection = NULL;
+	} else {
+		// XXX: https://github.com/libevent/libevent/issues/265
```

Comment:
This should be done only for `ENOTCONN` not always

Reply:
fixed